File: RSMK2/RSMK2/ProviderOperationImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RSMK2
{
    public class ProviderOperationImpl : MarshalByRefObject, ProviderOperation
    {
        public static List<Provider> lst = new List<Provider>();

        public List<Provider> GetAll() => lst;
        public List<Provider> DeleteAt(int index)
        {
            lst.RemoveAt(index);
            return lst;
        }
        public List<Provider> Add(Provider newItem)
        {
            lst.Add(newItem);
            return lst;
        }
    }
}

File: zwsserver/src/operation/ProviderOperationImpl.java
package operation;

import types.Provider;
import java.util.ArrayList;
import java.util.List;

public class ProviderOperationImpl implements ProviderOperation {

    public static List<Provider> lst = new ArrayList<Provider>();

    static {
        lst.add(new Provider(12, "quas", "soap"));
        lst.add(new Provider(132, "exort", "rocks"));
    }

    @Override
    public List<Provider> GetAll() {
        return lst;
    }

    @Override
    public List<Provider> DeleteAt(int index)
    {
        lst.remove(index);
        return lst;
    }

    @Override
    public List<Provider> Add(Provider newItem)
    {
        lst.add(newItem);
        return lst;
    }
}